syntax = "proto3";

import "mixing_state.proto";
import "ls_state.proto";

package bayesmix;

message MarginalState {
  message ClusterState {
    oneof val {
      UnivLSState univ_ls_state = 1;
      MultiLSState multi_ls_state = 2;
    }
  }

  repeated ClusterState cluster_states = 1;
  repeated int32 cluster_allocs = 2 [packed = true];
  repeated int32 cluster_cards = 3 [packed = true];
  repeated MixingState mixing_states = 4;
  int32 iteration_num = 5;
}
